framework/console/controllers/BaseMigrateController.php

```diff
@@ -752,7 +759,7 @@ protected function migrateDown($class)
     protected function createMigration($class)
     {
         $this->includeMigrationFile($class);
-        return new $class();
+        return new $class(['compact' => $this->compact]);
```

Comment:
This is unacceptable, it wil break migrations of 3rd party libraries such as `yii2-mongodb` as thier migrations have no `compact` property!

Reply:
Oh... right.

Reply:
Would the following be better?

```php
...
$migration = new $class();
if ($migration->canSetProperty('compact')) {
    $migration->compact = $this->compact;
}
return $migration;
```

Reply:
No.
Who can be sure Migration class is a descendant of `BaseObject`?
`BaseMigrationController::createMigration()` should be left as it was before. `compact` optnios should be applied only at `MigrationController::createMigration()`.